src/metabase/api/database.clj

```diff
@@ -808,13 +808,11 @@
                                         :engine       engine
                                         :details      details-or-error
                                         :is_full_sync is_full_sync
-                                        :is_on_demand (boolean is_on_demand)
+                                        :is_on_demand is_on_demand
                                         :cache_ttl    cache_ttl
                                         :creator_id   api/*current-user-id*}
-                                       (sync.schedules/schedule-map->cron-strings
-                                        (if (:let-user-control-scheduling details)
-                                          (sync.schedules/scheduling schedules)
-                                          (sync.schedules/default-randomized-schedule)))
```

Comment:
Orthogonal to your changes, but it feels strange that we re-randomize their schedule whenever this endpoint is hit, even if it already as non user defined.